### ISSUE TITLE
Add docker build to release pipeline 

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#2583]
Add a build of 'released' docker images to the release pipeline

ie create docker images of the form
odpi/egeria:1.4

The base build should now allow this configurability, but the actual build execution needs adding to the release pipeline.

Targeting 1.6 as we need a good run on addressing other release pipeline issues before adding more complexity

**Comment by planetf1** [comment]
We need to build our 'release' docker images when we run our release pipeline.

Whilst we could try and take a perfect 'staging' approach, a pragmatic solution would be to modify the existing release pipeline:

I would like to take the steps from our 'merge' job
```
         - task: DownloadSecureFile@1
            name: dockerConfig
            inputs:
              secureFile: odpi-docker.json
          - script: mv $(dockerConfig.secureFilePath) ${HOME}/.docker/config.json
          - task: Docker@2
            displayName: Login to Docker Hub
            inputs:
              command: login
              containerRegistry: odpi-dockerhub
```
And add them, along with required variables, to the release pipeline.

In addition, when building we should specify the following additions to the maven command line:
```
                 -Ddocker
                  -Ddocker.taglatest
                  -Ddocker.repo=odpi
                  -Ddocker.registry=docker.io
                  -Ddocker.images=core
                  -Ddockerfile.googleContainerRegistryEnabled=false

```

The result is that docker images would get published to dockerhub. It's possible they would need to be redone if we found problems, though rerunning the pipeline would do that, and the consuming apps we support (the compose yaml & k8s) wouldn't refer to them until the release was complete in any case.

@bramwelt I could give this a go, but am a little wary of the pipeline vs release differences (IMO it's an odd distinction in azure), would you be able to take a look?

Given we're about to release 1.5, targetting for 1.6 would be fine.

**Comment by planetf1** [comment]
Having looked at this a little more I have opted for a simple fix. The docker image for any egeria release (egeria-release-*) will be created and pushed as part of the existing merge build. 

This will be done the same as master, but to avoid introducing the issue fixed in #2137 a condition has been added on the publish to artifactory. Also we only tag the docker build as latest for master.

I did not use variables as I'm not familar enough with them in ap and wanted a low risk approach. This could be refactored in future to avoid duplication

Whilst this does mean a docker image may get published prematurely, and will always be 'best so far' on that release stream, the helm charts or docker compose yaml using them won't be published until the release as a whole is. 

The 'proper' solution would probably to to start publishing all docker images from the merge build - for master AND releases - to some alternative repo - or at least with an alternate name. Then we could have a stage of the release pipeline that pushed the final image. Ideally this would really just be a publish (move / copy) rather than a build. However this is more complicated.

I believe the current solution in the PR is appropriate for now.
